d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -821,9 +829,43 @@ public void getLoadBalancedServiceProperties(String serviceName, boolean waitFor
     }
   }
 
+  public void processGetLoadBalancedServicePropertiesTimeoutError(
```

Comment:
Not a super great name... Maybe `handleTimeoutFromGetProperties`?

Reply:
Good suggestion. Done